src/domains/User/UserSettings/components.tsx

```diff
@@ -57,38 +59,51 @@ export function TownhallUserSettings({ user }: { user: ClientSafeUser }) {
     const [state, setState] = React.useState(user.settings.townhall);
     const buildHandler = buildCheckboxUpdate<typeof state>(setState);
     // TODO: API Request
+
     return (
         <SettingsList>
-            <SettingsItem
-                helpText={text.townhall.anonymous}
-                name='Appear Anonymous'
-            >
-                <Switch
-                    checked={state.anonymous}
-                    onChange={buildHandler('anonymous')}
-                />
+            <SettingsItem helpText={text.townhall.anonymous} name='Appear Anonymous'>
+                <Switch checked={state.anonymous} onChange={buildHandler('anonymous')} />
             </SettingsItem>
         </SettingsList>
     );
 }
 
 export function NotificationSettings({ user }: { user: ClientSafeUser }) {
-    const [state, setState] = React.useState(user.settings.notifications);
-    const buildHandler = buildCheckboxUpdate<typeof state>(setState);
+    const [notifState, setNotifState] = React.useState(user.settings.notifications);
     // TODO: API Request
+    // done? look at lines 84-89
+    const [snack] = useSnack();
+    const changeNotifEnd = React.useCallback(() => API.changeNotif(notifState.enabled, notifState.types), [
+        notifState.enabled,
+        notifState.types,
+    ]);
+
+    // send user notif, BACKEND flips it, sends it back, logic happens there
+    // this does make it a tad slow to update visually (at least in storybook)
+    // but this makes sense as what if the backend cannot update the notification?
+    // then it updates visually here? no, it should only update IF it succeeds
```

Comment:
Good comment! 

 We can make this optimistic UI (first definition found online, and it's accurate). 
>Optimistic UI is a pattern that you can use to simulate the results of a mutation and update the UI even before receiving a response from the server. Once the response is received from the server, the optimistic result is thrown away and replaced with the actual result.

source: https://www.apollographql.com/docs/react/performance/optimistic-ui/  (we're gonna be using this package in the future, but it's irrelevant right now other than their definition of optimistic UI)

Reply:
Thank you for the link! 
To get this done, I figured the UI should change upon user click, and just "revert" and show a snack upon failure, telling the user to try again.
Let me know if I should change this approach, I am not sure this is correct, but it does work in storybook and no longer lags.